travel-event-host-app/src/components/auth-dialog/AuthDialog.tsx

```diff
@@ -52,16 +55,45 @@ export default function AuthDialog(props: AuthDialogProps) {
         return;
       }
       // TODO: submit the form
-      console.log('Signup form is valid');
+      const fetch = async () => {
+        const signUpRes = await clientSignUp(
+          formValues.name,
+          formValues.email,
+          formValues.password1,
+        );
+
+        if (signUpRes) {
+          if (signUpRes === true) {
+            setErrors({});
+            props.onDialogClose();
+            props.onRegisterSuccess;
+          } else {
+            setErrors(signUpRes as Record<string, string[]>);
+          }
+        }
+      };
+      fetch();
+      return;
     } else {
       // Validate the login form
       const loginValidationResult: Record<string, string[]> = validateLogin();
       if (Object.keys(loginValidationResult).length > 0) {
+        console.log(loginValidationResult);
         setErrors(loginValidationResult);
         return;
       }
+      const fetch = async () => {
```

Comment:
Same here - we could rename this to be something more descriptive

Reply:
Sure